### DISCUSSION TITLE
给”社区共建计划“提几个想法，让 EP 更好

### DISCUSSION THREAD

**Discussion by Otto-J** [element-plus/element-plus#4309] [Ideas]
各位，element-plus 目前还在高速迭代，在提供 vue3 优秀组件的基础上，不断引入的组件特性、新的构建流程和新的迭代计划。

作为忠实用户，提几个想法，欢迎大家讨论。

# 目的
让 EP 发展更有规划，关注 web开发者的使用感受和需求，让 EP 发展得更好。

# 受众
能看到这篇文章的 web 开发者，EP 粉丝

# 现状

考虑到预计年底的 statble 稳定版，这段时间 EP 还会持续迭代，迭代的过程都可以在 [release changelog](https://github.com/element-plus/element-plus/releases) 中看到，目前纳入工作流中自动完成的，不需要手工干预。目前 log 大致分三部分：新特性feature、问题修复Bugs、内部优化 Refactors。每一项操作都很重要，对细节感兴趣的，可以进入具体的 commit 观察代码变动，并参与讨论。

由此产生了几个想法。

# 想法

对于新功能 feature 可以在 changeLog 的基础上展开说明，比如解释组件功能的使用场景、api设计的思考。可以沉淀到具体组件的文档中，也可以先出更新说明 release post，每一项140字以内足矣。产生了内容可以进一步分发，比如发到前端社区中获得更多参与讨论，用户的反馈也有助于加强EP源于社区，亲近用户的信心。甚至也可以 EP weekly/monthly，或者 release email 等等。

如何具体去做。我可以关注最近一个迭代周期产生的 issues/PR/Dissussion ，已完成的内容进行评论和摘录，未结束的给出阶段反馈。

代码的变化都可以追溯，这一点已经做到了，还是来自 issues/PR/Dissussion。这一部分可以在 issues 中 pin 一个讨论。内容介绍 Projects 里的不同看板。我还是觉得目前 Projects 里的 Plan 太多了，迭代不够清楚。

如何具体去做。整理 Project 的看板，每个迭代周期发一次 Disscussion，PIn 到页面中，除了透明排期，也更容易让有余力的同学参与到组件共建中。

所有变更决策提供意见征询的窗口期，避免新特性带来意料之外的破坏性变化。Vue 有 RFC 来讨论某一项功能是否必要引入，组件库也可以。与其他组件库不同，EP源于社区，更亲近开发者，可以走到更前。

如何具体去做。这个看讨论吧，这个比较理想。

**Comment by CaiWenlie** [comment]
新功能不重要，赶紧出稳定版吧
